storm-client/test/jvm/org/apache/storm/topology/PersistentWindowedBoltExecutorTest.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.topology;
+
+import org.apache.storm.Config;
+import org.apache.storm.generated.GlobalStreamId;
+import org.apache.storm.state.KeyValueState;
+import org.apache.storm.streams.Pair;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.apache.storm.windowing.Event;
+import org.apache.storm.windowing.TimestampExtractor;
+import org.apache.storm.windowing.TupleWindow;
+import org.apache.storm.windowing.WaterMarkEvent;
+import org.apache.storm.windowing.WaterMarkEventGenerator;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static org.mockito.AdditionalAnswers.returnsArgAt;
+
+/**
+ * Unit tests for {@link PersistentWindowedBoltExecutor}
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class PersistentWindowedBoltExecutorTest {
+    public static final String LATE_STREAM = "late_stream";
+    private PersistentWindowedBoltExecutor<KeyValueState<String, String>> executor;
+    private IStatefulWindowedBolt<KeyValueState<String, String>> mockBolt;
+    private OutputCollector mockOutputCollector;
+    private TopologyContext mockTopologyContext;
+    private Map<String, Object> mockStormConf = new HashMap<>();
+    private TimestampExtractor mockTimestampExtractor;
+    private WaterMarkEventGenerator mockWaterMarkEventGenerator;
+    private KeyValueState mockPartitionState;
```

Comment:
Nit: Could probably avoid rawtypes complaints by using [at]Mock similar to what you've done with Captor